### PR TITLE
Navbar UI updates

### PR DESCRIPTION
**Pull request by thatbudakguy**
- Refactor templates to keep Spotlight's header navbar
- Hide the sign in link in the navbar

Ref #1372. First commit makes it so that the navbar still appears on auth pages.

### CODE REVIEW

File: app/assets/stylesheets/dlme.scss [hunk 1]
```diff
@@ -169,7 +169,11 @@ body {
 }
 
 .masthead .site-title {
-  white-space: normal;
+  visibility: hidden;
+
+  + small {
```
Comment (jcoyne) [inline app/assets/stylesheets/dlme.scss:174]: What is the purpose of this?  What content is in the small that we want hidden?  Can we just put the hidden style on that element in the template so it's easier to understand the connection?

File: app/assets/stylesheets/dlme.scss [hunk 2]
```diff
@@ -202,6 +206,11 @@ body {
   }
 }
 
+.image-masthead .background-container,
+.image-masthead .background-container-gradient {
+  top: 0;
```
Comment (jcoyne) [inline app/assets/stylesheets/dlme.scss:211]: What is the purpose of this?

File: app/assets/stylesheets/dlme.scss [hunk 3]
```diff
@@ -253,28 +262,34 @@ body {
 }
 
 .topbar {
+  position: absolute;
+  width: 100vw;
```
Comment (jcoyne) [inline app/assets/stylesheets/dlme.scss:266]: Why is these necessary? Topbar is already full width.

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/dlme.scss (hunk 1 above)
Comment (thatbudakguy) [reply app/assets/stylesheets/dlme.scss:174]: Spotlight by default inserts the exhibit title and some other content, but we don't want to show it because DLME has only one exhibit and the title is rendered elsewhere. I think this rendering happens in spotlight, so we'd need to override an entire template to do that. Part of the point of this PR was reducing the amount of things we override.

File: app/assets/stylesheets/dlme.scss (hunk 3 above)
Comment (thatbudakguy) [reply app/assets/stylesheets/dlme.scss:266]: The existing styles don't have an effect once its position is changed to `absolute`, which is necessary to overlay it on the masthead.

File: app/assets/stylesheets/dlme.scss (hunk 2 above)
Comment (thatbudakguy) [reply app/assets/stylesheets/dlme.scss]: Tested without it and it seemed to work, so I've removed this.

File: app/assets/stylesheets/dlme.scss (hunk 1 above)
Comment (thatbudakguy) [reply app/assets/stylesheets/dlme.scss:175]: Added a comment to note the above

File: app/assets/stylesheets/dlme.scss (hunk 3 above)
Comment (jcoyne) [reply app/assets/stylesheets/dlme.scss:262]: Why are we changing to absolute? I'm really struggling to understand that purpose of these changes.

Comment (thatbudakguy) [reply app/assets/stylesheets/dlme.scss:262]: We overrode several spotlight templates in order to construct an HTML setup where the navbar was inside the masthead, in order to implement the current design. This PR gets rid of all the overrides in favor of creating that appearance via a small amount of CSS instead, so we don't diverge from Spotlight so much. It also means that we get to keep the default navbar, so it automatically appears on pages that do not have a masthead, like login pages, which is part of the ticket.

Comment (thatbudakguy) [reply app/assets/stylesheets/dlme.scss:262]: Specifically, we use absolute so that the navbar can be positioned over the top of the masthead, which is its following sibling.

File: app/assets/stylesheets/dlme.scss [hunk 4]
```diff
@@ -386,4 +400,13 @@ body {
   dd, dl {
     margin-bottom: 0.25rem;
   }
-}
\ No newline at end of file
+}
+
+// Turn off transparency for the topbar on pages where there's no masthead
```
Comment (jcoyne) [inline app/assets/stylesheets/dlme.scss:405]: Can you add  "(e.g. the pages provided by Devise)"

File: app/assets/stylesheets/dlme.scss [hunk 5]
```diff
@@ -253,17 +258,18 @@ body {
...
-  @include media-breakpoint-down('sm') {
-    margin-bottom: 0;
+  // extra spacing when there's no masthead in between navbar and content
+  + #main-container {
+    padding-top: 8rem !important;
   }
 }
 
-.masthead .topbar .navbar-nav {
+.topbar  .navbar-nav {
```
Comment (jcoyne) [inline app/assets/stylesheets/dlme.scss:272]: Remove extra space

File: app/assets/stylesheets/dlme.scss [hunk 6]
```diff
@@ -340,8 +351,12 @@ body {
     border: 0;
   }
 
-  .masthead .site-title {
-    text-align: center;
+  .masthead .site-title-container {
+    display: none;
```
Comment (jcoyne) [inline app/assets/stylesheets/dlme.scss:355]: Can you add a comma here with the reason we're not displaying it?

File: app/assets/stylesheets/dlme.scss [hunk 7]
```diff
@@ -275,6 +281,11 @@ body {
       text-decoration: underline;
     }
   }
+
+  // hide the "sign in" link
+  &[href="/users/sign_in"] {
```
Comment (jcoyne) [inline app/assets/stylesheets/dlme.scss:286]: Can we remove this and just not render the link https://github.com/sul-dlss/dlme/blob/main/app/views/shared/_user_util_links.html.erb#L41-L45

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/dlme.scss (hunk 4 above)
Comment (thatbudakguy) [reply app/assets/stylesheets/dlme.scss]: added

File: app/assets/stylesheets/dlme.scss (hunk 5 above)
Comment (thatbudakguy) [reply app/assets/stylesheets/dlme.scss]: fixed

File: app/assets/stylesheets/dlme.scss (hunk 6 above)
Comment (thatbudakguy) [reply app/assets/stylesheets/dlme.scss:351]: added

File: app/assets/stylesheets/dlme.scss (hunk 7 above)
Comment (thatbudakguy) [reply app/assets/stylesheets/dlme.scss]: Yeah, good call. made that change

**Review by jcoyne — APPROVED** [review]
